### PR TITLE
matrix-corporal: init at 2.0.1

### PR DESCRIPTION
**Pull request by D4ndellion**
###### Motivation for this change
matrix-corporal is a proxy and reconciliator for matrix-synapse letting you allow/deny API endpoints, and automatically join users to rooms/communities according to a configuration policy

###### Things done

Only adds a package, I might make a module in the future, but settings and secrets was a little hard in #93777 

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dali99** [comment]
@Ma27 @aanderse from the last issue

**Comment by aanderse** [comment]
I only ever reviewed the NixOS module and not the package... Sorry I don't have more to offer :frowning_face:

**Comment by Ma27** [comment]
Wouldn't it make sense to add a module as well though?

**Comment by Infinisil** [comment]
Build works, but running it I get:
```
---------------------------------------------------------- [ Version:  ]
GitCommit:
GitBranch:
GitState:
GitSummary:
BuildDate:
```

I think it would be a good idea to try to make sure this includes at least the git commit and/or version (not the build date though, because that's impure).

**Comment by SuperSandro2000** [comment]
@dali99 please address the review feedback.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 99074` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>matrix-corporal</li>
  </ul>
</details>

File: pkgs/servers/matrix-corporal/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchFromGitHub {
+    owner = "devture";
+    repo = pname;
+    rev = version;
+    sha256 = "0xmfdqvyl956s2y0qa7mcbkbcdzpicjxxxaz181gwdin8h9mdj0p";
+  };
+
+  buildFlagsArray = [
+    "-ldflags=-X main.GitCommit=${version} -X main.GitBranch=${version} -X main.GitState=nixpkgs -X main.GitSummary=${version} -X main.Version=${version}"
```
Comment (SuperSandro2000) [inline pkgs/servers/matrix-corporal/default.nix:15]:
```suggestion
    "-ldflags=-s -w -X main.GitCommit=${version} -X main.GitBranch=${version} -X main.GitState=nixpkgs -X main.GitSummary=${version} -X main.Version=${version}"
```

Comment (dali99) [reply pkgs/servers/matrix-corporal/default.nix:15]: Do we always strip debug information in nixpkgs?

I just copied the ones from https://nixos.org/manual/nixpkgs/unstable/#ssec-go-common-attributes which doesn't

**Comment by dali99** [comment]
The package has been updated to 2.0.0 and now prints:
```
                 _        _                                                _
 _ __ ___   __ _| |_ _ __(_)_  __      ___ ___  _ __ _ __   ___  _ __ __ _| |
| '_ \ _ \ / _\ | __| '__| \ \/ /____ / __/ _ \| '__| '_ \ / _ \| '__/ _\ | |
| | | | | | (_| | |_| |  | |>  <_____| (_| (_) | |  | |_) | (_) | | | (_| | |
|_| |_| |_|\__,_|\__|_|  |_/_/\_\     \___\___/|_|  | .__/ \___/|_|  \__,_|_|
                                                    |_|
---------------------------------------------------------- [ Version: 2.0.0 ]
GitCommit: 2.0.0
GitBranch: 2.0.0
GitState: nixpkgs
GitSummary: 2.0.0
BuildDate: 
```
Im not sure if that's what we want, and I took the liberty of using the GitState to say it is built with nixpkgs

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 99074` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>matrix-corporal</li>
</ul>
</details>

File: pkgs/servers/matrix-corporal/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{lib, fetchFromGitHub, buildGoModule }:
```
Comment (SuperSandro2000) [inline pkgs/servers/matrix-corporal/default.nix:1]:
```suggestion
{ lib, fetchFromGitHub, buildGoModule }:
```

**Pull request merged**